src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java

```diff
@@ -223,6 +230,86 @@ public CreateCaseEventResult createCaseEvent(final String caseReference, final C
             .build();
     }
 
+    public CreateCaseEventResult createCaseSystemEvent(final String caseReference,
```

Comment:
Most of the logic in this method is re-used from the existing one, probably worth splitting out common to private method(s)?